frontend/src/pages/NewQuestion/index.tsx

```diff
@@ -41,9 +47,21 @@ const NewQuestion = () => {
   const [uploadedImagesUrl, setUploadedImagesUrl] = useState<string[]>([]);
   const [isPreviewQuestion, setIsPreviewQuestion] = useState<boolean>(false);
 
-  const [pythonTemplate, setPythonTemplate] = useState<string>("");
-  const [javaTemplate, setJavaTemplate] = useState<string>("");
-  const [cTemplate, setCTemplate] = useState<string>("");
+  const [testCases, setTestCases] = useState<TestCase[]>([
+    { id: uuidv4(), input: "", expectedOutput: "" },
+  ]);
+  const [testcaseInputFile, setTestcaseInputFile] = useState<File | null>(null);
+  const [testcaseOutputFile, setTestcaseOutputFile] = useState<File | null>(
+    null
+  );
+
+  const [codeTemplates, setCodeTemplates] = useState<{ [key: string]: string }>(
+    {
+      python: "",
+      java: "",
+      c: "",
+    }
```

Comment:
maybe can consider adding a default template for every question? 

    {
      python: '# Please do not modify the main function\ndef main():\n    print(convert_to_string_format(solution()))\n\n\n# Write your code here\ndef solution():\n    return None\n\n\nif __name__ == "__main__":\n    main()\n',
      java: "public class Main {\n    // Please do not modify the main function\n    public static void main(String[] args) {\n        System.out.println(convert_to_string_format(solution()));\n    }\n\n    // Write your code here and return the appropriate type\n    public static String solution() {\n        return null;\n    }\n}\n",
      c: '#include <stdio.h>\n\n// Write your code here and return the appropriate type\nconst char* solution() {\n    return "";\n}\n\n// Please do not modify the main function\nint main() {\n    printf("%s\\n", convert_to_string_format(solution()));\n    return 0;\n}\n',
    }